File: impprog/zhgyak/feladat2/Py/feladat2Fug.py


def keres(lista,keresett):
    #print(lista)
    letezik=True
    megtalalt=False
    id=int(len(lista)/2)
    previd=0

    while(not(megtalalt) and letezik):
        if id == previd:
            letezik=False
        
        if lista[id] == keresett:
            megtalalt=True
        elif lista[id] < keresett:
            id=int((id+len(lista)-1)/2)
        else:
            id=int((0+id)/2)

    
    print(id)

File: impprog/zhgyak/feladat2/C/feladat2Fug.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int keres(int tomb[],int keres,int len){
    int id=-1;
    bool found=false;
    bool exists=true;
    //printf("%d", len);
    id=len/2;
    int previd=0;
    do{
        if(previd==id){
            exists=false;
        }
        if(keres==tomb[id]){
            found=true;
        } else if(keres > tomb[id] && id<len){
            id=(len+id-1)/2;
            //printf("%d, %d\n", id,previd);
        } else{
            id=(0+id)/2;
            //printf("%d, %d\n", id,previd);
        }

    } while(!found && exists);
    
    if(found)
        printf("%d\n", id);
    if(!exists)
        printf("%d\n",-1);
}